scripts/fabricstat

```diff
@@ -0,0 +1,203 @@
+        """
+        full_table_id = FABRIC_PORT_STATUS_TABLE_PREFIX + port_name
+        oper_state = self.db.get(self.db.STATE_DB, full_table_id, FABRIC_PORT_STATUS_FIELD)
+        if oper_state is not None:
+            return oper_state
+        return STATUS_NA
+
+    def get_counters(self, counter_bucket_dict, table_id):
+        fields = ["0"] * len(counter_bucket_dict)
+        for pos, counter_name in counter_bucket_dict.items():
+            full_table_id = COUNTER_TABLE_PREFIX + table_id
+            counter_data =  self.db.get(self.db.COUNTERS_DB, full_table_id, counter_name)
+            if counter_data is None:
+                 fields[pos] = STATUS_NA
+            elif fields[pos] != STATUS_NA:
+                fields[pos] = str(int(fields[pos]) + int(counter_data))
+        return fields
+
+    def get_cnstat(self):
+        """
+        Get the counters info from database.
+        """
+        assert False, 'Need to override this method'
+
+    def cnstat_print(self, cnstat_dict, errors_only=False):
+        """
+        Print the counter stat.
+        """
+        assert False, 'Need to override this method'
+
+FABRIC_COUNTERS_PORT_NAME_MAP = "FABRIC_COUNTERS_PORT_NAME_MAP"
+
+PortStat = namedtuple("PortStat", "in_cell, in_octet, out_cell, out_octet, crc, fec_correctable, fec_uncorrectable, symbol_err")
+port_counter_bucket_list = [
+    'SAI_PORT_STAT_IF_IN_FABRIC_DATA_UNITS',
+    'SAI_PORT_STAT_IF_IN_OCTETS',
+    'SAI_PORT_STAT_IF_OUT_FABRIC_DATA_UNITS',
+    'SAI_PORT_STAT_IF_OUT_OCTETS',
+    'SAI_PORT_STAT_IF_IN_ERRORS',
+    'SAI_PORT_STAT_IF_IN_FEC_CORRECTABLE_FRAMES',
+    'SAI_PORT_STAT_IF_IN_FEC_NOT_CORRECTABLE_FRAMES',
+    'SAI_PORT_STAT_IF_IN_FEC_SYMBOL_ERRORS',
+    ]
+port_counter_bucket_dict = {k : v for k, v in enumerate(port_counter_bucket_list)}
+
+portstat_header_all = ['ASIC', 'PORT', 'STATE',
+                       'IN_CELL', 'IN_OCTET', 'OUT_CELL', 'OUT_OCTET',
+                       'CRC', 'FEC_CORRECTABLE', 'FEC_UNCORRECTABLE', 'SYMBOL_ERR']
+portstat_header_errors_only = ['ASIC', 'PORT', 'STATE',
+                               'CRC', 'FEC_CORRECTABLE', 'FEC_UNCORRECTABLE', 'SYMBOL_ERR']
+
+class FabricPortStat(FabricStat):
+    def get_cnstat(self):
+        counter_port_name_map = self.db.get_all(self.db.COUNTERS_DB, FABRIC_COUNTERS_PORT_NAME_MAP)
+        cnstat_dict = OrderedDict()
+        if counter_port_name_map is None:
+            return cnstat_dict
+        for port_name in natsorted(counter_port_name_map):
+            cntr = self.get_counters(port_counter_bucket_dict, counter_port_name_map[port_name])
+            cnstat_dict[port_name] = PortStat._make(cntr)
+        return cnstat_dict
+
+    def cnstat_print(self, cnstat_dict, errors_only=False):
+        if len(cnstat_dict) == 0:
+            print("Counters %s empty" % self.namespace)
+            return
+
+        table = []
+        header = None
+        asic = int(self.namespace[len(ASIC_NAME_PREFIX):])
```

Comment:
We can use this function defined in sonic_py_common
[`get_asic_id_from_name(asic_name):` ](https://github.com/Azure/sonic-buildimage/blob/935923b8059a48b01a4daa765cdfa4bb6560562a/src/sonic-py-common/sonic_py_common/multi_asic.py#L136)